### PR TITLE
Update button usage guidelines

### PR DESCRIPTION
**Pull request by ashygee**
This PR updates our current [button usage guidelines](https://primer.style/design/ui-patterns/button-usage) and with more specific guidelines and updated screenshots.

### CODE REVIEW

File: content/ui-patterns/button-usage.mdx [hunk 1]
```diff
@@ -2,116 +2,92 @@
...
-    <Caption>Don't use all-caps or other text formats</Caption>
-  </Dont>
-</DoDontContainer>
+## Types and usage
 
-### Keep button text concise
+### Primary
 
-When designing buttons, keep the text as short but descriptive as possible. Buttons are to be understood by their type and text; if further clarity is required, consider adding an octicon or adding a text description.
+<img width="188" alt="Primary button has a green background and white text" src="https://user-images.githubusercontent.com/10384315/102147130-19a3b700-3e1f-11eb-8581-3b58b053dadd.png">
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:23]:
```suggestion
<img width="188" alt="Primary button has a green background and white text" src="https://user-images.githubusercontent.com/10384315/102147130-19a3b700-3e1f-11eb-8581-3b58b053dadd.png" />
```

File: content/ui-patterns/button-usage.mdx [hunk 2]
```diff
@@ -2,116 +2,92 @@
...
 
-### Adaptable full-width
+Use for the main action on a page or view. Use only one per page or view.
 
-In certain scenarios, it may make sense to use full-width buttons that extend to the width of their container. Use this style sparingly, but some appropriate use-cases include:
+### Secondary
 
-- Modals with a singular call-to-action
-- Responsive behavior of buttons when scaled down
+<img width="188" alt="Secondary button has light gray background and dark gray text" src="https://user-images.githubusercontent.com/10384315/102147132-19a3b700-3e1f-11eb-9cc9-0c7a6fdb8e4a.png">
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:29]:
```suggestion
<img width="188" alt="Secondary button has light gray background and dark gray text" src="https://user-images.githubusercontent.com/10384315/102147132-19a3b700-3e1f-11eb-9cc9-0c7a6fdb8e4a.png" />
```

File: content/ui-patterns/button-usage.mdx [hunk 3]
```diff
@@ -2,116 +2,92 @@
...
-    <Caption>Don't misuse this modifier by extending buttons full-width unnecessarily</Caption>
-  </Dont>
-</DoDontContainer>
+Use for general actions on the page or view.
 
-### Use text descriptions
+### Outline
 
-In certain scenarios, buttons require supporting text to give the user more context. Use text labels sparingly, as buttons should generally be able to be understood on their own.
+<img width="188" alt="Outline button has light gray background and blue text" src="https://user-images.githubusercontent.com/10384315/102147129-19a3b700-3e1f-11eb-94b7-941c140701e3.png">
```
Comment (colebemis) [inline content/ui-patterns/button-usage.mdx:35]:
```suggestion
<img width="188" alt="Outline button has light gray background and blue text" src="https://user-images.githubusercontent.com/10384315/102147129-19a3b700-3e1f-11eb-94b7-941c140701e3.png" />
```

**Review by colebemis — COMMENTED** [review]
You might be getting errors because `img` tags need to be self-closing (i.e. `` not ``)

**Review by yaili — CHANGES_REQUESTED** [review]
I think the **Anatomy**, and each of the sections in **Variations** should have an image.

**Comment by colebemis** [comment]
Closing this since there are a lot of conflicts with @mperrotti's recent update to the button docs. @ashygee feel free to reopen if there are still changes you'd like to make.

**Pull request closed without merging**